src/keyboard/Keymap.tsx

```diff
@@ -19,7 +20,7 @@ export interface KeymapProps {
   scale: LayoutZoom;
   selectedLayerIndex: number;
   selectedKeyPosition: number | undefined;
-  onKeyPositionClicked: (keyPosition: number) => void;
+  onKeyPositionClicked: Dispatch<SetStateAction<number | undefined>>
```

Comment:
for my own understanding, what's the advantage of this Dispatch type?

Reply:
it's just to satisfy the type

```
// normal usage
onKeyPositionClicked(0)

// with just normal fn, this will be incorrect
onKeyPositionClicked((prev) => prev + 1)
```